### ISSUE TITLE
Repeating groups: Kunne sette opp logikk for å skjule en eller flere rader i en repeterende gruppe

### ISSUE THREAD

**Issue by rvessb** [Altinn/app-frontend-react#389]

**Comment by olemartinorg** [comment]
Kan dette fungere? https://docs.altinn.studio/app/development/ux/fields/grouping/alternatives/#filter
Ref tråd på Slack: https://altinn.slack.com/archives/C02EVE4RU82/p1665734097717869 (hvor det ble notert at det fungerer, men ikke så veldig bra)

**Comment by xmrsa** [comment]
Her har jeg opprettet en bug på slack-saken: https://github.com/Altinn/app-frontend-react/issues/545 i håp om at dette kan være en mulighet for løsning, i alle fall på kort sikt.

**Comment by rvessb** [comment]
Beskrevet løsning tillater kun sjekk mot en verdi på et bestemt felt.  Vi trenger i tillegg støtte for:
- Større og mindre enn funksjon
- Inneholder/ inneholder ikke (der variabelen det sjekkes mot er  en kommasep. liste)
- And, Or, Nor, Nand - funksjoner mellom kriteriene

**Comment by olemartinorg** [comment]
Det dere ønsker er beskrevet i denne issuen (som nok vil deles opp etterhvert): https://github.com/Altinn/altinn-studio/issues/8799
Med unntak av nor og nand, riktignok. Men det er enkelt nok å implementere i expressions-motoren om det trengs.

**Comment by olemartinorg** [comment]
Ja, herlighet - her har jeg jo ikke tenkt meg om.. :facepalm: Godt at noen andre er mer våkne! Fint å ta med punktene som er nevnt i den når det kommer til implementering, men ellers lukker jeg den.

**Comment by olemartinorg** [comment]
Copying over the description and tasks from the closed #584, and adding to it:

### Description

It should be possible to use expressions to hide an entire row inside a repeating group. This functionality should deprecate the existing [filters for repeating groups](https://docs.altinn.studio/app/development/ux/fields/grouping/alternatives/#filter), which has been deemed too inflexible.

### Functional requirements
- [ ] It should be possible to use expressions to hide an entire row of a repeating group, including using lookups into components/the data model for the specific row. It might be possible to use the first node inside the group row as the `LayoutNode` when evaluating the expression.
- [ ] Hiding the entire row should also mark every component inside it as hidden, and the row should not render in the table at all. This includes nested groups.
- [ ] Updating the data used to evaluate the expression should immediately hide/unhide the row(s)
- [ ] Hidden rows should not produce any validation errors/messages for any components inside it
- [ ] Hidden rows should not be displayed in summaries/confirmations/PDF

### Other tasks
- [ ] Add the new property needed to control this into the JsonSchema for layouts
- [ ] Implement support for this in `app-lib-dotnet` as well (or create issues for extending the expression engine to properly hide components inside hidden rows of the repeating group) 
  - The rows in the data model should remain as-is, but should possibly be removed when completing the instance and submitting the final data (this behaviour should possibly be configurable).

### Additional Information

- #540 
- #569 
- #558

**Comment by olemartinorg** [comment]
Another example of usage: Having a MultipleSelect/Checkboxes/etc component that is used to prefill a repeating group structure. If you uncheck/unselect a value in the MultipleSelect/Checkboxes/etc, you'll currently have to delete data from the later repeating group. However, if a filter could be set to only show rows in the repeating group matching the selected prefill items, all the repeating groups could be prefilled by default (and only shown when selected). This also means that going back to re-show an earlier group preserves the data you had filled in there.
